src/webview/java/org/medicmobile/webapp/mobile/RDToolkitSupport.java

```diff
@@ -0,0 +1,232 @@
+		ProvisionMode provisionMode = !rdtFilter.trim().matches("\\S+") ? ProvisionMode.CRITERIA_SET_AND : ProvisionMode.CRITERIA_SET_OR;
+		Intent intent = RdtIntentBuilder
+				.forProvisioning()
+				.setCallingPackage(ctx.getPackageName())
+				.setReturnApplication(ctx)
+				// Type of RD Test to choose from
+				.requestProfileCriteria(rdtFilter, provisionMode)
+				// Unique ID for RD Test
+				.setSessionId(sessionId)
+				// First line text to display in RDToolkit App and to differentiate running tests
+				.setFlavorOne(patientName)
+				// Second line text to display in RDToolkit App and to differentiate running tests
+				.setFlavorTwo(patientId)
+				.setCloudworksBackend(monitorApiURL, patientId)
+				.build();
+
+		if (intent.resolveActivity(ctx.getPackageManager()) != null) {
+			ctx.startActivityForResult(intent, RDTOOLKIT_PROVISION_ACTIVITY_REQUEST_CODE);
+		}
+
+		return intent;
+	}
+
+	Intent captureRDTest(String sessionId) {
+		Intent intent = RdtIntentBuilder
+				.forCapture()
+				// Unique ID for RD Test
+				.setSessionId(sessionId)
+				.build();
+
+		if (intent.resolveActivity(ctx.getPackageManager()) != null) {
+			ctx.startActivityForResult(intent, RDTOOLKIT_CAPTURE_ACTIVITY_REQUEST_CODE);
+		}
+
+		return intent;
+	}
+
+	String getImage(String path) {
+		if (path == null || path.length() == 0) {
+			return null;
+		}
+
+		try {
+			trace(this, "RDToolkitSupport :: Retrieving image file");
+			Uri filePath = getUriFromFilePath(path);
+			ParcelFileDescriptor parcelFileDescriptor = ctx
+					.getContentResolver()
+					.openFileDescriptor(filePath, "r");
+
+			InputStream file = new FileInputStream(parcelFileDescriptor.getFileDescriptor());
+			Bitmap imgBitmap = BitmapFactory.decodeStream(file);
+			file.close();
+
+			trace(this, "RDToolkitSupport :: Compressing image file");
+			ByteArrayOutputStream outputFile = new ByteArrayOutputStream();
+			imgBitmap.compress(Bitmap.CompressFormat.JPEG, 75, outputFile);
+
+			trace(this, "RDToolkitSupport :: Encoding image file to Base64");
+			byte[] imageBytes = outputFile.toByteArray();
+			String imageEncode = Base64.encodeToString(imageBytes, Base64.NO_WRAP);
+
+			return imageEncode;
+
+		} catch (Exception exception) {
+			error(exception, "RDToolkitSupport :: Failed to process image file from path: %s", path);
+		}
+
+		return null;
+	}
+
+//> PRIVATE HELPERS
+
+	private String processCaptureResponseActivity(int resultCode, Intent intentData) {
+		if (resultCode != RESULT_OK) {
+			throw new RuntimeException("RDToolkitSupport :: Bad result code for capturing result: " + resultCode);
+		}
+
+		try {
+			JSONObject response = parseCaptureResponseToJson(intentData);
+			return makeCaptureResponseJavaScript(response);
+
+		} catch (Exception exception) {
+			error(exception, "RDToolkitSupport :: Problem serialising the captured result");
+			return safeFormat("console.log('Problem serialising the captured result: %s')", exception);
```

Comment:
As above, if we log these as errors, is a feedback doc created?

Reply:
Done!